lib/mutant/mutator/node/regexp.rb

```diff
@@ -0,0 +1,104 @@
+module Mutant
+  class Mutator
+    class Node
+
+      # Generic mutator
+      module RegularExpression
+        class Parser
+          def self.parse(source)
+            Node.new(Regexp::Parser.parse(source, 'ruby/2.3'))
+          end
+
+          class Node < DelegateClass(Regexp::Expression::Base)
+            def to_source
+              to_s
+            end
+
+            def with(expressions)
+              new_node =
```

Comment:
I realize you didn't ask for review, but I figured I'd mention this anyway. This code is a bit inconsistent with code in Node#dispatch. In that code you explicitly `dup`, then mutate the object then use the object elsewhere. I don't mind using `tap` like this, but consistency is probably more important, so I would write this as:

```ruby
def with(expression)
  new_node = dup
  new_node.expressions = expressions
  self.class.new(new_node)
end
```